xonsh/main.py

```diff
@@ -113,7 +113,7 @@ def path_argument(s):
                     metavar='args',
                     help='Additional arguments to the script specified '
                          'by script-file',
-                    nargs='*',
+                    nargs=argparse.REMAINDER,
```

Comment:
> this strangely didn't produce any errors

That is probably because of the amalgamation and another file importing argparse. This should still be fixed though.

>  Xonsh atm doesn't seem to be able to start in interactive mode when a file is passed @scopatz

What do mean by start?  Does it run the file?  Xonsh should still be able to run files in interactive mode, even though it won't pop up with a command line prompt.  This behaviour seems to work just fine for me on master right now.